reference/matrix/dense_kernels.cpp

```diff
@@ -622,32 +622,31 @@ void calculate_max_nnz_per_row(std::shared_ptr<const ReferenceExecutor> exec,
             std::max(num_nonzeros, num_stored_elements_per_row);
     }
 
-    *result = num_stored_elements_per_row;
+    result = num_stored_elements_per_row;
 }
 
 GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(
     GKO_DECLARE_DENSE_CALCULATE_MAX_NNZ_PER_ROW_KERNEL);
 
 
 template <typename ValueType>
-void calculate_nonzeros_per_row(std::shared_ptr<const ReferenceExecutor> exec,
-                                const matrix::Dense<ValueType>* source,
-                                Array<size_type>* result)
+void count_nonzeros_per_row(std::shared_ptr<const ReferenceExecutor> exec,
+                            const matrix::Dense<ValueType>* source,
+                            size_type* result)
```

Comment:
In case we don't have access to the underlying array. But this is more of a preference, since it requires unpacking to a pointer only in a single place (core) instead of all kernel implementations.